webpack/assets/javascripts/react_app/components/MemoryAllocationInput/MemoryAllocationInput.js

```diff
@@ -0,0 +1,136 @@
+import React, { useCallback, useState } from 'react';
+import PropTypes from 'prop-types';
+import { WarningTriangleIcon, ErrorCircleOIcon } from '@patternfly/react-icons';
+import { HelpBlock, Grid, Col, Row } from 'patternfly-react';
+import { translate as __ } from '../../common/I18n';
+import NumberInput from '../common/forms/NumberInput';
+import './memoryAllocationInput.scss';
+import { noop } from '../../common/helpers';
+
+const MemoryAllocationInput = ({
+  label,
+  defaultValue,
+  onChange,
+  maxValue,
+  recommendedMaxValue,
+}) => {
+  const GBFormat = 'GB';
+  const MBFormat = 'MB';
+  const GBStep = 1024;
+  const MBStep = 256;
+  const [value, setValue] = useState(defaultValue);
+
+  const getValidationState = useCallback(() => {
+    if (value > maxValue && maxValue != null) return 'error';
+    else if (value > recommendedMaxValue && recommendedMaxValue != null)
+      return 'warning';
+    return null;
+  }, [recommendedMaxValue, maxValue, value]);
+
+  const handleValueIncrease = () => {
+    if (value >= GBStep) {
+      setValue(value + GBStep);
+    } else {
+      setValue(value + MBStep);
+    }
+  };
+
+  const handleValueDecrease = () => {
+    if (value <= GBStep) {
+      setValue(value - MBStep);
+    } else {
+      setValue(value - GBStep);
+    }
+  };
+
+  const handleTypedValue = v => {
+    if (v > GBStep) {
+      setValue(Math.round(v / GBStep) * GBStep);
+    } else {
+      setValue(Math.round(v / MBStep) * MBStep);
+    }
+  };
+
+  const handleChange = v => {
+    if (v === value + 1) {
+      handleValueIncrease();
+    } else if (v === value - 1) {
+      handleValueDecrease();
+    } else {
+      handleTypedValue(v);
+    }
+    onChange(value);
+  };
+
+  const format = v => {
+    // used the buttons
+    if (v % MBStep === 0) {
+      if (v >= GBStep) {
+        return `${v / GBStep} ${GBFormat}`;
+      }
+
+      return `${v} ${MBFormat}`;
+    }
+
+    // typed value
+    return v;
+  };
+
+  const parser = str => str.replace(/\D/g, '');
+  const validationState = getValidationState();
+  return (
+    <Grid>
+      <Row>
+        <Col>
+          <NumberInput
+            value={value}
+            format={format}
+            parser={parser}
+            onChange={v => handleChange(v)}
```

Comment:
you can drop the redundant anonymous function
```suggestion
            onChange={handleChange}
```